### PR TITLE
ParkFind and Filter pages

### PR DESCRIPTION
**Pull request by mccullochk**
## Screenshots
</kbd> </kbd> </kbd>

## Notes
- There are a few missing SVG icons that we are still waiting on
- A lot of the parks still don't have images
- Some of the urls provided for images don't work and show as a blank image
- Resolves [pe1130](https://app.clubhouse.io/parks/story/1130/filter-page)
- Resolves [pe1447](https://app.clubhouse.io/parks/story/1447/convert-png-park-amenity-icons-to-svg-icons)
- Resolves [pe1399](https://app.clubhouse.io/parks/story/1399/parkfind-page)

### CODE REVIEW

**Comment by mccullochk** [comment]
@chadfawcett, this is a big change. Sorry about that. I'd primarily love a second set of eyes on the filter functions please!

Also, I did tweak the `<ParkList />` component slightly, to get the text to wrap, to display the special characters, and to not be bolded text because that's the way it appears in Figma. If any of these changes are in appropriate let me know and I will change it back.

File: src/utils/helpers.js [hunk 1]
```diff
@@ -12,27 +13,97 @@ export function getClosestParksByAmenityTypeAndID(
...
+    searchTerm !== undefined
+      ? parks.filter((park) =>
+          park.searchableTitle.toLowerCase().includes(searchTerm.toLowerCase())
+        )
+      : parks
+
+  /*
+   * Further filter parks based on distance, activity, and facility filters
+   */
+  const filterDistance = distance >= 99 ? maxDistance : distance
```
Comment (chadfawcett) [inline src/utils/helpers.js]: `Math.min()` is a little more readable to me for calculations like this.
```suggestion
  const filterDistance = Math.min(distance, maxDistance)
```

**Review by chadfawcett — APPROVED** [review]
@mccullochk the changes look good! Thanks for updating the ParkList component.

As for the filter functions I feel like there a couple ways you could go. The approach you took is likely the more performant way, but I think it makes it less readable. Since the number of parks isn't a lot (as far as a computer is concerned at least) and I don't think the number of parks will grow exponentially, I'd suggest maybe going with a less performant route if it will help the readability. What I'm thinking is that you could filter on each criteria separately. Something along the lines of:

```javascript
let filteredParks = parks

if (searchTerm)
  filteredParks = filteredParks.filter(/* filter based on search term*/)

if (location)
  filteredParks = filterParks.filter(/* filter based on location/distance)

if (selectedActivityIDs.length > 0)
  // filter by activities

if (selectedFacilitiesByIDs.length > 0)
  // filter by facilities

return filteredParks
```

I _think_ this will be more readable, and easier to add more filtering criteria to in the future if needed. It also helps get rid of some redundant checks like when you're checking the selected length and setting the matches boolean to true, when it's already true to begin with.

I think what is there is fine. Just my 2 cents on readability.

File: src/utils/helpers.js (hunk 1 above)
Comment (mccullochk) [reply src/utils/helpers.js]: I understand why `Math.min()` is your preferred approach for readable calculation. I agree however that is not what I'm trying to achieve here. I want to use the provided filter distance until that distance reaches 100km at which point we basically want to remover the filter distance or "max it out" which is what I'm doing here. After 100km the distance filter would return all parks. Does that make sense?

**Comment by mccullochk** [comment]
@chadfawcett, thanks for your honest and helpful feedback. I like having it setup for easy inclusion of additional filter. How does the readability look to you now?

File: src/utils/helpers.js (hunk 1 above)
Comment (chadfawcett) [reply src/utils/helpers.js]: Oh sorry, I was going off memory and thought `maxDistance` was `100`. I guess I mixed that up with the `defaultDistanceFilter`.

LGTM 👍

**Review by chadfawcett — APPROVED** [review]
Much easier to follow now, thanks!

**Pull request merged**